src/cudadev/DataFormats/TrackerTopology.h

```diff
@@ -0,0 +1,592 @@
+#ifndef TRACKERTOPOLOGY_H
+#define TRACKERTOPOLOGY_H
+
+#include "DataFormats/DetId.h"
+#include "DataFormats/PixelSubdetector.h"
+#include "DataFormats/SiStripEnums.h"
+
+#include <vector>
+#include <string>
+
+//knower of all things tracker geometry
+//flexible replacement for PXBDetId and friends
+//to implement
+// endcap pixel
+
+class TrackerTopology {
```

Comment:
I wonder why `TrackerTopology` is needed? I did notice the added `#include` in `SiPixelRawToClusterGPUKernel.cu` but by quick look on the diff the reason wasn't evident (e.g. the full). If only a subset of this header is needed, could the copy be limited to that?

Reply:
Good point, I'll check if it can be reduced.

Reply:
OK, the only part that uses the topology is https://github.com/cms-patatrack/pixeltrack-standalone/blob/fb9033e3101d51e0bab142a276c92876237ab2dc/src/cudadev/plugin-SiPixelClusterizer/SiPixelRawToClusterGPUKernel.cu#L57

so I can copy those three constants and avoid including the full topology.

Reply:
The `PixelSubdetector::PixelBarrel` is defined in `PixelSubdetector.h` (added in this PR), and the `DetId::*` in `DetId.h` (added in this PR), so maybe the `TrackerTopology` could be avoided completely?

Reply:
Thanks for the suggestion, I've update the PR dropping `DataFormats/TrackerTopology.h` and including the other headers directly.